core/controller/src/main/scala/org/apache/openwhisk/core/controller/BasicAuthenticationDirective.scala

```diff
@@ -87,10 +139,25 @@ object BasicAuthenticationDirective extends AuthenticationDirectiveProvider {
 
   def identityByNamespace(
     namespace: EntityName)(implicit transid: TransactionId, system: ActorSystem, authStore: AuthStore) = {
-    Identity.get(authStore, namespace)
+    //Identity.get(authStore, namespace)
+    implicit val ec = authStore.executionContext
+    implicit val logging = authStore.logging
+    Identity.get(authStore, namespace) map { result =>
+      val blacklist = getOrCreateBlacklist
+      if (!blacklist.isEmpty && blacklist.isBlacklisted(
+            Try(result.authkey.asInstanceOf[BasicAuthenticationAuthKey].account).getOrElse(""))) {
+        Identity(
+          subject = result.subject,
+          namespace = result.namespace,
+          authkey = result.authkey,
+          rights = result.rights,
+          limits = UserLimits(invocationsPerMinute = Some(0), concurrentInvocations = Some(0), firesPerMinute = Some(0)))
+      } else result
+    }
   }
 
   def authenticate(implicit transid: TransactionId,
+                   system: ActorSystem,
```

Comment:
You can get the actor system with 
`extractActorSystem.flatMap { implicit system =>`
like in the` IAMAuthenticationDirective` and do not have to pass it

Reply:
Adapted according to the change request:
```
  def authenticate(implicit transid: TransactionId,
                   authStore: AuthStore,
                   logging: Logging): AuthenticationDirective[Identity] = {
    extractActorSystem.flatMap { implicit system =>
      extractExecutionContext.flatMap { implicit ec =>
        basicAuth(validateCredentials)
      }
    }
  }
```